source/_posts/Docker-Swarm.md

```diff
@@ -0,0 +1,240 @@
+
+<img src="/images/Docker-Swarm-overview.png" />
+Een overzicht hoe het cluster eruit ziet met 3 managers en 2 workers.
+
+Omdat ik Docker Swarm gebruik kan ik de volgende punten afvinken van mijn lijstje:
+- De oplossing moet schaalbaar zijn en bestand zijn tegen het uitvallen of offline gaan van servers. 
+    -- In het cluster zitten 3 managers dus er kan er 1 offline gaan. Ook hebben we 2 workers welke de containers kunnen hosten
+- Ik wil de oplossing kunnen hosten bij een Cloud provider, Hosting provider of lokaal om te kunnen testen.
+    -- Omdat Docker Swarm standaard in elke Docker installatie zit kan het zonder verdere installatie van tools gebruikt worden overal waar je Docker hebt geïnstalleerd.  
+
+Wat is Let's Encrypt
+---
+Let's Encrypt is een certificaatautoriteit opgericht op 16 april 2016. Het geeft X.509 certificaten uit voor het Transport Layer Security (TLS) encryptie-protocol, zonder dat dit kosten met zich meebrengt. De certificaten worden uitgegeven via een geautomatiseerd proces dat is ontworpen om het tot nu toe complexe proces van handmatige validatie, ondertekening, installatie en hernieuwing van certificaten voor beveiligde websites te elimineren. (Wikipedia)
+
+Voor meer informatie over Let's Encrypt zie https://letsencrypt.org/
+
+Wat is Traefik
+---
+Traefik is een opensource router welke speciaal is ontworpen voor container oplossingen. Traefik wordt als global service op elke manager gedeployed op het cluster. Dit wil zeggen elke node met als rol manager krijgt een Traefik container. De reden dat Traefik op de manager nodes gedeployed dient te worden is dat de Docker api wordt uitgelezen. Zodra er een container bij komt en deze is geconfigureerd met de Traefik labels kan Traefik de labels van de container uitlezen en een virtuele host aanmaken voor de container en een SSL-certificaat aanvragen bij Let's Encrypt. Zodoende is de container beschikbaar voor de buitenwereld met een SSL-certificaat.
+
+Voor meer informatie over Traefik zie https://traefik.io/
+
+Omdat ik Traefik gebruik als reverse proxy kan ik de volgende punten afvinken van mijn lijstje:
+- Er moeten meerdere web applicaties op kunnen draaien welke op poort 80 en 443 benaderd kunnen worden.
+    -- Doordat Traefik de labels van containers kan uitlezen maakt het automatisch virtuele hosts aan.
+- Alle webapplicaties moeten draaien op ssl, dit moet me zo min mogelijk werk kosten.
+    -- Traefik ondersteund out of the box Let's Encrypt SSL-certificaten.
+
+Wat is Portainer
+---
+Portainer is eeen opensource web interface om je Docker te beheren en om nieuwe containers uit te rollen en te updaten en bevat nog veel meer functionaliteiten. Ik ga hier niet in diepte behandelen wat Portainer allemaal kan voor meer informatie over portainer kun je terecht op https://www.portainer.io/. 
```

Comment:
> Ik ga hier niet...
Dit klinkt raar, je bedoelt eigenlijk: Het gaat in deze blog te ver om de ins en outs van Portainer te behandelen. Of iets dergelijks.